app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainFragment.kt

```diff
@@ -85,18 +85,9 @@ class MainFragment : Fragment() {
      */
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
-        if (resultCode == RESULT_OK) {
-            if (requestCode == AUTH_REQUEST_CODE) {
-                user = FirebaseAuth.getInstance().currentUser
-            }
+        if (resultCode == RESULT_OK && requestCode == AUTH_REQUEST_CODE) {
+            user = FirebaseAuth.getInstance().currentUser
```

Comment:
Condensed these nested if() statements into one using && https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/and.html

Reply:
I'd keep these separate.  onActivityResult is invoked from any startActivityForResult() call.  Each result will need to see if the resultCode is RESULT_OK, so best to put that in an outer if test, and then consider each different requestCode case in an inner if test.  Less typing, and a bit easier to read.